Add section about clearing the sessions store during OMERO.web upgrade

sbesson opened the pull request:
See https://github.com/ome/omero-web/pull/435, https://github.com/ome/omero-web/issues/459 and https://github.com/ome/omero-web/issues/383 for more context about the requirement for this new optional section
See also https://docs.djangoproject.com/en/3.2/topics/http/sessions/ for some reference documentation about the different types of Django session store backends.

Motivated by some of the changes introduced in OMERO.web 5.18.0. It is expected the same issues will arise when switching to Django 4.2 and this could happen anytime if the session serialization format is modified.

A new paragraph about common issues when OMERO.server fails to start is also added to the troubleshooting section and this PR should create a link that can be reference from UPGRADING.md in the source code repo.

sbesson committed:
Add section about clearing the sessions stored during OMERO.web upgrade

Motivated by some of the changes introduced in OMERO.web 5.18.0. It is
expected the same issues will arise when switching to Django 4.2 and
this could happen anytime if the session serialization format is modified.

A new paragraph about common issues when OMERO.server fails to start is
also added to the troubleshooting section and this PR should create a
link that can be reference from UPGRADING.md in the source code repo.

sbesson committed:
Fix markup for cleanup commands

sbesson committed:
Use program markup

will-moore commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,47 @@ All official OMERO.web plugins can be installed from PyPI_.
+  OMERO.web will use file-based sessions. The session files are stored under a
+  temporary folder determined by the the output of `tempfile.gettempdir()`,
+  usually `/tmp` and prefixed either with with the value of
+  :property:`omero.web.session_cookie_name` if the property if set of by
```
typos: ...if the property *is* set *or* by... - Also duplicate `with` in previous line

will-moore commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,47 @@ All official OMERO.web plugins can be installed from PyPI_.
+  usually `/tmp` and prefixed either with with the value of
+  :property:`omero.web.session_cookie_name` if the property if set of by
+  `sessionid` by default.
+  For a default configuration, the the following command will
```
duplicate `the`

will-moore commented on omero/sysadmins/troubleshooting.rst:
```diff
@@ -408,6 +408,15 @@ more details. There are a few known possibilities:
    if all of the prerequisites were installed from
    :doc:`OMERO.web deployment <unix/install-web/web-deployment>`.
 
+OMERO.web fails with start
```
maybe "with start" -> "to start" ?

sbesson committed:
Fix typos

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,47 @@ All official OMERO.web plugins can be installed from PyPI_.
+- if :property:`omero.web.session_engine` is either unset or set to
+  `omeroweb.filesessionstore` or `django.contrib.sessions.backends.file`,
+  OMERO.web will use file-based sessions. The session files are stored under a
+  temporary folder determined by the the output of `tempfile.gettempdir()`,
```
```suggestion
  temporary folder determined by the output of `tempfile.gettempdir()`,
```

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,47 @@ All official OMERO.web plugins can be installed from PyPI_.
+  OMERO.web will use file-based sessions. The session files are stored under a
+  temporary folder determined by the the output of `tempfile.gettempdir()`,
+  usually `/tmp` and prefixed either by the value of
+  :property:`omero.web.session_cookie_name` if the property if set of by
```
```suggestion
  :property:`omero.web.session_cookie_name` if the property is set of by
```

sbesson committed:
Rename troubleshooting section

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,47 @@ All official OMERO.web plugins can be installed from PyPI_.
+- if :property:`omero.web.session_engine` is set to
+  `django.contrib.sessions.backends.cache`, OMERO.web uses cached sessions
+  with a cache backend configured via :property:`omero.web.caches`. For a
+  Redis cache backend, sessions are stored using keys prefixed with
```
Add link to Redis?

sbesson replied:
Happy to do add a link. While writing this PR, I realised we had no reference section about the support of Redis as a backend (it's mostly mentioned as part of the installation workflow). Are you thinking of linking to upstream Redis? or django-redis?

Related this discussion, this PR makes the authoritative assumption that files and Redis cache are the primary supported backends for storing sessions. Looking into the code, there are other backends that may be configured including database-backed sessions or Memcached cache https://docs.djangoproject.com/en/3.2/topics/http/sessions/. A potential follow-up of this PR might be to expand on the list of session engines and describe their level of support.

sbesson committed:
Apply suggestions from code review

Co-authored-by: jean-marie burel <j.burel@dundee.ac.uk>

will-moore commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,47 @@ All official OMERO.web plugins can be installed from PyPI_.
+  OMERO.web will use file-based sessions. The session files are stored under a
+  temporary folder determined by the output of `tempfile.gettempdir()`,
+  usually `/tmp` and prefixed either by the value of
+  :property:`omero.web.session_cookie_name` if the property is set of by
```
of -> or

sbesson replied:
```suggestion
  :property:`omero.web.session_cookie_name` if the property is set or by
```

sbesson committed:
Update omero/sysadmins/omeroweb-upgrade.rst

jburel commented:
The failure is not related to the change. It started to happen on other repositories. I fixed the issue in https://github.com/ome/omero-guide-python/pull/58 
I will go over the repositories using rtd

jburel commented:
@sbesson you will have to rebase the PR in order to fix the error

sbesson committed:
Merge remote-tracking branch 'origin/develop' into omeroweb_clearcache

sbesson committed:
Add Redis link

sbesson commented:
This should ready for review and integration in the next documentation release. With the upcoming move to Django 4.2, I expect the session clearing step will be mandatory (unless people have switched to the JSON serializer)

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,48 @@ All official OMERO.web plugins can be installed from PyPI_.
+  usually `/tmp` and prefixed either by the value of
+  :property:`omero.web.session_cookie_name` if the property is set or by
+  `sessionid` if the property is unset.
+  For a default configuration, the following command will
```
maybe indicate earlier that by default it is unset
something like ``` :property:`omero.web.session_engine` (default: unset)```

sbesson replied:
This will be true also if `omero.web.session_engine` is explicitly set to a file backend.

By default configuration, I was rather thinking to the expectation that `SESSION_COOKIE_NAME` and possibly other Django envvars would not be overriden. Maybe I'll invert the two sentence here i.e. first give the command that should work in 95% of the cases then mention the variables that could modify the location of the temporary sessions

jburel replied:
The way it is written at the moment made one think that the default is unset.
Swapping the order should avoid the confusion

sbesson committed:
Reorder the section about clearing sections

Clarify the two main supported session backends and their configuration
Provide the example for the default configuration in both case
Add final sentence about additional backend configuration

sbesson committed:
readthedocs: fail on Sphinx warnings

sbesson commented:
See https://omero--2312.org.readthedocs.build/en/2312/sysadmins/omeroweb-upgrade.html#clear-the-sessions-store-optional for the updated text

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,58 @@ All official OMERO.web plugins can be installed from PyPI_.
+
+The process for clearing the session store depends on the storage backend:
+
+- OMERO.web session data is stored on the filesystem if the
```
data is plural. We have a mix of "data is" and "data are" in our doc.
We will need to fix that

sbesson replied:
Oh dear, I didn't mean to enter such an unchartered territory. To my defense, the Django documentation is extensively use of `session data` with the singular form of the verb. 😄 
I think I'll just use `sessions` instead of `session data` everywhere.

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -87,6 +87,58 @@ All official OMERO.web plugins can be installed from PyPI_.
+  if the :property:`omero.web.session_engine` property is set to
+  `django.contrib.sessions.backends.cache` and Redis is configured via the
+  :property:`omero.web.caches` property by setting the `BACKEND` to
+  'django.core.cache.backends.redis.RedisCache' and `LOCATION` to
```
```suggestion
  `django.core.cache.backends.redis.RedisCache` and `LOCATION` to
```

sbesson committed:
session data -> sessions

sbesson committed:
Use verbatim form for RedisCache backend value

jburel commented on omero/sysadmins/omeroweb-upgrade.rst:
```diff
@@ -121,11 +121,11 @@ The process for clearing the session store depends on the storage backend:
+  `django.core.cache.backends.redis.RedisCache` and `LOCATION` to
   the URL of the Redis instance.
 
   Session data is stored as key/value pairs in the Redis store with the name of
```
``Sessions are stored as key/value`` should also be adjusted to ``Sessions are..`` in that case.
Same for ``Session data is stored under a temporary folder```

sbesson committed:
More session data renaming
